processor/k8sprocessor/kube/client.go

```diff
@@ -235,6 +236,30 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 			tags[r.Name] = c.extractField(v, r)
 		}
 	}
+
+	if c.Rules.HostName {
+		// Basing on v1.17 Kubernetes docs, when a hostname is specified, it takes precedence over
+		// the associated metadata name, see:
+		// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
+		if pod.Spec.Hostname == "" {
+			tags[conventions.AttributeHostName] = pod.Name
+		} else {
+			tags[conventions.AttributeHostName] = pod.Spec.Hostname
+		}
```

Comment:
```golang
hostname :=  pod.Spec.Hostname
if hostname == "" {
  hostname = pod.Name
}
tags[conventions.AttributeHostName] = hostname
```

Reply:
Any particular reasons for this? (other than maybe accessing `pod.Spec.Hostname` just once)

Reply:
@pmalek-sumo, I think such approach makes the intention more clear. Currently, you check `pod.Spec.Hostname` and assign `pod.Name` which is not obvious why, unless the line below is being read

Of course, that might be a small thing, but one where clarity could be improved easily

Reply:
👍🏻  done